api/test/unit/views/media_views_test.py

```diff
@@ -0,0 +1,40 @@
+from test.factory.models.image import ImageFactory
+from unittest import mock
+from urllib.error import HTTPError
+
+from rest_framework.test import APIClient
+
+import pytest
+
+from catalog.api.models.image import Image
+
+
+@pytest.fixture
+def api_client() -> APIClient:
+    return APIClient()
+
+
+@pytest.fixture
+def image() -> Image:
+    return ImageFactory.create()
+
+
+@pytest.mark.django_db
+def test_thumb_error(api_client, image):
+    error = None
+
+    def urlopen_503_response(url, **kwargs):
+        nonlocal error
+        error = HTTPError(url, 503, "Bad error upstream whoops", {}, None)
+        raise error
```

Comment:
Why does this need to be set to `None` initially, then set to `HTTPError` within the function? Is that to test that the mock's side effect does indeed get called? I imagine we could just declare the `error` at the beginning of the test and use it throughout.

Reply:
The error needs the `url` that is passed to the `urlopen` function, so it can't be initialized outside of this handler.

Reply:
Ah! Got it, thanks 🙂